DependencyInjection/DoctrineExtension.php

```diff
@@ -85,6 +87,14 @@ protected function dbalLoad(array $config, ContainerBuilder $container)
     {
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
         $loader->load('dbal.xml');
+        $chainLogger = $container->getDefinition('doctrine.dbal.logger.chain');
+        $logger      = new Reference('doctrine.dbal.logger');
+        if (! method_exists(SQLParserUtils::class, 'getPositionalPlaceholderPositions') && method_exists(LoggerChain::class, 'addLogger')) {
+            // doctrine/dbal < 2.10.0
+            $chainLogger->addMethodCall('addLogger', [$logger]);
+        } else {
+            $chainLogger->addArgument([$logger]);
```

Comment:
this must be removed. You are adding an argument here with an array with 1 element, and then later adding a second argument with an array with 2 elements. But the ChainLogger expects the list of loggers as its first argument so that's why things don't work.

Reply:
an alternative would be to use `replaceArgument` in the ChildDefinition

Reply:
That might be better :thinking: in case people are using the abstract service definition to build their custom service.

Reply:
Pushed again, this time with `replaceArgument`

Reply:
if they use it to build their own custom service, they will need to change their code for the DBAL 3 way anyway.

Reply:
Yes, but I'm switching to the constructor as soon as possible, not as late as possible. This means people using DBAL >=2.10 and building their own custom service should need this.

Reply:
> Pushed again, this time with `replaceArgument`

We could even go a cleaner way to prepare for the day we will require DBAL 2.10+ in the bundle (hopefully soon, if the PR adding back PHP 7.1 support in 2.12 is accepted):
- stop using `ChildDefinition` in `loadDbalConnection` but a normal `Definition` using the chainlogger class (this requires adding the `addLogger` call for `$logger` in the branch dealing with DBAL 2.9, but that's fine)
- mark the `doctrine.dbal.logger.chain` abstract service as deprecated (in case someone was using it for their own service)

Reply:
That does seem more straightforward, let me try 👍